src/frontends/tensorflow/src/op/depthwise_conv_2d.cpp

```diff
@@ -14,66 +14,55 @@ namespace tensorflow {
 
-    Strides ng_strides(2);
-    Strides ng_dilations(2);
+    Strides strides(2);
+    Strides dilations(2);
+    convert_nhwc_to_hw(is_nhwc, tf_strides, strides);
+    convert_nhwc_to_hw(is_nhwc, tf_dilations, dilations);
+
     Shape ng_image_shape(2);
     Shape ng_kernel_shape(2);
 
-    convert_nhwc_to_hw(is_nhwc, ng_input.get_shape(), ng_image_shape);
-    convert_nhwc_to_hw(is_nhwc, tf_strides, ng_strides);
-    convert_nhwc_to_hw(is_nhwc, tf_dilations, ng_dilations);
-    convert_nhwc_to_nchw(is_nhwc, ng_input);
-
-    auto& ng_filter_shape = ng_filter.get_shape();
-    ng_kernel_shape[0] = ng_filter_shape[0];
-    ng_kernel_shape[1] = ng_filter_shape[1];
-
-    CoordinateDiff ng_padding_below;
-    CoordinateDiff ng_padding_above;
-    make_padding(tf_padding_type,
-                 ng_image_shape,
-                 ng_kernel_shape,
-                 ng_strides,
-                 ng_dilations,
-                 ng_padding_below,
-                 ng_padding_above);
-
-    // H W I M -> H W I 1 M
-    auto filter_shape = make_shared<Constant>(
-        element::u64,
-        Shape{5},
-        ov::Shape{ng_filter_shape[0], ng_filter_shape[1], ng_filter_shape[2], 1, ng_filter_shape[3]});
-    auto reshaped_filter = make_shared<Reshape>(ng_filter, filter_shape, false);
-
-    // H W I 1 M -> I M 1 H W
-    auto order = make_shared<Constant>(element::i64, Shape{5}, vector<int64_t>{2, 4, 3, 0, 1});
-    auto transposed_filter = make_shared<opset8::Transpose>(reshaped_filter, order);
+    convert_nhwc_to_nchw(is_nhwc, input, ov::Rank(4));
 
-    auto ng_conv_node = make_shared<GroupConvolution>(ng_input,
-                                                      transposed_filter,
-                                                      ng_strides,
-                                                      ng_padding_below,
-                                                      ng_padding_above,
-                                                      ng_dilations);
-    auto ng_conv = ng_conv_node->output(0);
+    // prepare filter to have a number of groups equal to CIN
+    auto unsqueeze_filter =
+        make_shared<Unsqueeze>(filter, make_shared<Constant>(element::i64, Shape{1}, std::vector<int64_t>{3}));
+    auto transposed_filter =
+        make_shared<Transpose>(unsqueeze_filter,
+                               make_shared<Constant>(element::i64, Shape{5}, std::vector<int64_t>{2, 4, 3, 0, 1}));
 
-    convert_nchw_to_nhwc(is_nhwc, ng_conv);
-    set_node_name(node.get_name(), ng_conv.get_node_shared_ptr());
-    return {ng_conv};
+    ov::Output<ov::Node> group_conv = make_shared<GroupConvolution>(input,
+                                                                    transposed_filter,
+                                                                    strides,
+                                                                    CoordinateDiff({}),
+                                                                    CoordinateDiff({}),
```

Comment:
Same as for AvgPool, is there really no padding needed here? If we currently adding support of conb without padding, please add comment to add support for padding later.

Reply:
`AvgPool` does not have explicit padding that is why they are empty

Reply:
`AvgPool` does not have explicit padding either.